hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieStreamingSink.scala

```diff
@@ -247,4 +285,18 @@ class HoodieStreamingSink(sqlContext: SQLContext,
       writeClient = Option.empty
     }
   }
+
+  private def canSkipBatch(batchId: Long): Boolean = {
+    // get the latest checkpoint from the commit metadata to check if the microbatch has already been prcessed or not
+    val lastCommit = metaClient.get.getActiveTimeline.getCommitsTimeline.filterCompletedInstants().lastInstant()
+    if (lastCommit.isPresent) {
+      val commitMetadata = HoodieCommitMetadata.fromBytes(
+        metaClient.get.getActiveTimeline.getInstantDetails(lastCommit.get()).get(), classOf[HoodieCommitMetadata])
+      val lastCheckpoint = commitMetadata.getMetadata(SinkCheckpointKey)
+      if (!StringUtils.isNullOrEmpty(lastCheckpoint)) {
+        latestBatchId = lastCheckpoint.toLong
+      }
+    }
+    latestBatchId >= batchId
```

Comment:
so we deserialize the batch id from storage. how can we guarantee is always monotonically increasing?  users can start a completely new streaming job to the same table, then will the ids become not comparable?

Reply:
Yes this is not meant to work for multiple writers with multiple spark checkpoint locations for the same table.

Reply:
ok in multi-writer case, not sure the current commit metadata in form of single KV pair would suffice future design requirements. we may need a more flexible structure like k={a:b}? just want to avoid handling BWC later

Reply:
+1 Might be good to make the data model support multiple values from day 1